digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala

```diff
@@ -0,0 +1,377 @@
+package fi.liikennevirasto.viite
+
+import fi.liikennevirasto.digiroad2.{GeometryUtils, Point, Vector3d}
+import fi.liikennevirasto.digiroad2.asset.{BoundingRectangle, SideCode}
+import fi.liikennevirasto.digiroad2.asset.SideCode.{AgainstDigitizing, BothDirections, TowardsDigitizing}
+import fi.liikennevirasto.digiroad2.util.Track
+import fi.liikennevirasto.viite._
+import fi.liikennevirasto.viite.dao.Discontinuity._
+import fi.liikennevirasto.viite.dao.LinkStatus._
+import fi.liikennevirasto.viite.dao._
+import fi.liikennevirasto.viite.process.TrackSectionOrder
+
+object ProjectValidator {
+
+  sealed trait ValidationError {
+    def value: Int
+    def message: String
+    def notification: Boolean
+  }
+
+  object ValidationError {
+    val values = Set(MinorDiscontinuityFound, MajorDiscontinuityFound, InsufficientTrackCoverage, DiscontinuousAddressScheme,
+      SharedLinkIdsExist, NoContinuityCodesAtEnd, UnsuccessfulRecalculation, MissingEndOfRoad)
+    //Viite-942
+    case object MissingEndOfRoad extends ValidationError {def value = 0
+      def message = MissingEndOfRoadMessage
+      def notification = false}
+    //Viite-453
+    //There must be a minor discontinuity if the jump is longer than 0.1 m (10 cm) between road links
```

Comment:
maybe instead of values we should have variable location where value can be change so comment doesn't have to be maintained?